tests/test_tree.py

```diff
@@ -91,6 +91,22 @@ def test_sitemap_tree_for_homepage(self):
                 """.format(base_url=self.TEST_BASE_URL)).strip(),
             )
 
+            m.get(
+                self.TEST_BASE_URL + '/robots.txt',
+                headers={'Content-Type': 'text/plain'},
+                text=textwrap.dedent("""
+                    User-agent: *
+                    Disallow: /whatever
+                    Crawl-delay: 5
```

Comment:
Possibly this test could also *test* for whether the delay actually happens. The easiest way (but not the best perhaps) would be to set the test `Crawl-Delay` to, say, 2 seconds, and then confirm that the sitemap fetcher code didn't return results before those two seconds.